packages/dds/tree/src/test/time/tableTree.spec.ts

```diff
@@ -0,0 +1,646 @@
+					title: `Undo insert the middle column and row ${count} times`,
+					tableSize: treeSize,
+					cellValue,
+					setupOperation: (tree: TableTreeDefinition) => {
+						const { Column, Row, table } = tree;
+						for (let i = 0; i < count; i++) {
+							const column = new Column({ id: `column-${uuid()}` });
+							const row = new Row({ id: `row-${uuid()}`, cells: {} });
+							table.insertColumn({ index: Math.floor(table.columns.length / 2), column });
+							table.insertRow({ index: Math.floor(table.rows.length / 2), row });
+						}
+					},
+					stackOperation: (tree: TableTreeDefinition) => {
+						const { undoStack } = tree;
+						// assert.equal(stack.undoStackLength, count);
+						for (let i = 0; i < 2 * count; i++) {
+							undoStack.pop()?.revert();
+						}
+					},
+					maxBenchmarkDurationSeconds,
+				});
+
+				// Test the execute time of the SharedTree for redoing an insert row and a column in the middle for a given number of times.
+				runUndoRedoBenchmark({
+					title: `Redo insert the middle column and row ${count} times`,
+					tableSize: treeSize,
+					cellValue,
+					setupOperation: (tree: TableTreeDefinition) => {
+						const { Column, Row, table, undoStack } = tree;
+						for (let i = 0; i < count; i++) {
+							const column = new Column({ id: `column-${uuid()}` });
+							const row = new Row({ id: `row-${uuid()}`, cells: {} });
+							table.insertColumn({ index: Math.floor(table.columns.length / 2), column });
+							table.insertRow({ index: Math.floor(table.rows.length / 2), row });
+						}
+						for (let i = 0; i < 2 * count; i++) {
+							undoStack.pop()?.revert();
+						}
+					},
+					stackOperation: (tree: TableTreeDefinition) => {
+						const { redoStack } = tree;
+						// assert.equal(stack.undoStackLength, count);
+						for (let i = 0; i < 2 * count; i++) {
+							redoStack.pop()?.revert();
+						}
+					},
+					maxBenchmarkDurationSeconds,
+				});
+			});
+		}
+
+		// Set/Remove-related tests that are limited by treeSize
+		for (const count of validRemoveCounts) {
+			describe(`Column Removal`, () => {
+				// Test the execute time of the SharedTree for removing a column in the middle for a given number of times.
+				runBenchmark({
+					title: `Remove a column in the middle ${count} times`,
+					tableSize: treeSize,
+					cellValue,
+					operation: (tree: TableTreeDefinition) => {
+						const { table } = tree;
+						for (let i = 0; i < count; i++) {
+							table.removeColumn(`column-${i}`);
+						}
+					},
+					maxBenchmarkDurationSeconds,
+				});
+
+				// Test the execute time of undoing remove a column in the middle for a given number of times.
+				runUndoRedoBenchmark({
+					title: `Undo remove the middle column ${count} times`,
+					tableSize: treeSize,
+					cellValue,
+					setupOperation: (tree: TableTreeDefinition) => {
+						const { table } = tree;
+						for (let i = 0; i < count; i++) {
+							table.removeColumn(`column-${i}`);
```

Comment:
Column removal is currently a little funky in the table APIs (something we will be working to improve over the coming months). For now, it only removes the column node from the tree, while the cells associated with that node (which live in under the row) remain in the tree. This is different from how SharedMatrix works, and isn't representative of the case we want to capture in these tests. Check out the documentation on the `TableSchema` namespace - they provide a recommended pattern for implementing column removal that also removes the associated cells. I would recommend adding a similar function somewhere in this file that removes the columns and cells together in a transaction and use that in any tests that do column removal.

Reply:
Note: the performance of this case will likely not be very good (and will be much worse than the same case in SharedMatrix). This is a known issue that we'll be working on this semester. Having these tests in place will help validate the performance fixes as we make them.